osquery/filesystem/filesystem.cpp

```diff
@@ -139,20 +140,23 @@ Status listFilesInDirectory(const boost::filesystem::path& path,
     boost::filesystem::directory_iterator begin_iter(path);
     boost::filesystem::directory_iterator end_iter;
     for (; begin_iter != end_iter; begin_iter++) {
-      if (!boost::filesystem::is_directory(begin_iter->path())) {
-        results.push_back(begin_iter->path().string());
+      try {
+        if (boost::filesystem::is_regular_file(begin_iter->path())) {
+          results.push_back(begin_iter->path().string());
+        }
+      } catch (const boost::filesystem::filesystem_error& e) {
+        if (ignore_error == 0) {
```

Comment:
I guess, but the return status doesn't reveal much information about the failure. It'll be hard to:
* try to do some filesystem operation
* backoff from an error
* iterate and discover erroring files/nodes/etc
* do something about those errors
* loop

If there's no optional parameter to get the erroring files, I'd say the expectation is to operate as best as possible. Since an error here may include a subdirectory is not searchable by the user.